### DISCUSSION TITLE
Writing fixed_size_list column?

### DISCUSSION THREAD

**Discussion by bertt** [G-Research/ParquetSharp#327] [General]
Hi, I've got a Parquet file (see zipped attachment region_points.parquet) with a column 'geometry' of type fixed_size_list.
When opening the file in https://parquetreader.com/ the schema looks like:

![image](https://user-images.githubusercontent.com/538812/210364754-7463f9f5-4e06-409d-abe2-553347f2130e.png)

Question is how can I write to such a column type?

I've tried something with:

```
var columns = new Column[]
{
    new Column<Double?[]>("geometry")
};

var geometryWriter = rowGroup.NextColumn().LogicalWriter<Double?[]> ();
 ```

But that gives:

![image](https://user-images.githubusercontent.com/538812/210365146-263228af-c172-467a-9d58-a9db4fe855c1.png)

[region_points.zip](https://github.com/G-Research/ParquetSharp/files/10336861/region_points.zip)

**Comment by adamreeve** [comment]
Hi @bertt 

The fixed size list type isn't part of the Parquet specification, but is an [Arrow columnar format type](https://arrow.apache.org/docs/format/Columnar.html#fixed-size-list-layout). The example file you've attached contains an embedded Arrow schema that provides additional typing information that can be used when reading the file into the Arrow columnar format, and it's here that the fixed size list type is specified. There's some more information on this at https://arrow.apache.org/docs/cpp/parquet.html#roundtripping-arrow-types.

ParquetSharp doesn't write Arrow format data so it's not currently possible to add the Arrow schema easily. If you really wanted you could probably figure out how to serialize an IPC schema message and include it in your Parquet file metadata though. There's code in the dotnet Apache.Arrow package that does this but it's an internal implementation detail and it looks like it's not possible to just serialize a schema by itself: https://github.com/apache/arrow/blob/6a936c4ff5007045e86f65f1a6b6c3c955ad5103/csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs#L637

**Reply by bertt** [reply]
Hi Adam, ok thanks good to know it's not something trivial :-) Maybe I'll do some experiments with the Apache.Arrow package.
